System/Metrics/Prometheus.hs

```diff
@@ -400,33 +404,79 @@ ofAll _ = Metric
 
 -- $registering-and-deregistering
 -- Before metrics can be sampled, they need to be registered with the
--- metric store. Once registered, metrics can also be deregistered.
+-- metric store.
 
 ------------------------------------------------------------------------
 -- ** Registering
 
 -- $registering
--- Metrics are identified by both their metric class and label set. If
--- you try to register a metric at an identifier that is already in use
--- by an existing metric, the existing metric will be deregistered and
--- replaced by the new metric.
+-- Metrics are registered as either "permanent" or "removable".
+-- Removable metrics can later be deregistered and replaced, while
+-- permanent metrics cannot.
 --
--- Upon `register`ing a set of metrics, you will be given a handle that
--- can be used to deregister the newly registered metrics /specifically/,
--- in the following sense. If a deregistration handle targets a metric,
--- and that metric is replaced by a new metric, the new metric will not
--- be deregistered if the handle is handle used.
-
--- | Atomically register one or more metrics. Returns a handle for
--- atomically deregistering those metrics specifically. Throws
--- 'ValidationError' if the given registration contains any invalid
--- metric or label names.
-register
+-- Metrics are identified by the combination of their metric name and
+-- label set. If you try to register a metric at an identifier that is
+-- already in use by an existing metric, what happens depends on whether
+-- the metrics are permanent or removable. If both metrics are
+-- removable, then the existing metric will be deregistered and replaced
+-- by the new metric; in all other cases, an exception will be thrown.
+--
+-- Upon registering a set of removable metrics, you will be given a
+-- handle that can be used to deregister the newly registered metrics
+-- /specifically/, in the following sense. If a deregistration handle
+-- targets a metric, and that metric is replaced by a new metric, the
+-- new metric will not be deregistered if the handle is handle used.
```

Comment:
Not sure I understand this part of the comment `if the handle is handle used` is that suppose to be `if the handle is already used`?